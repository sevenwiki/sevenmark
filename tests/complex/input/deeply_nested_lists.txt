{{{#list #1 #style="margin-left:20px"
[[**Chapter 1**: Introduction]]
[[Subsection: {{{#list #a [[Point A with *emphasis*]] [[Point B with [now]]] }}} ]]
[[**Chapter 2**: Advanced Topics]]
[[Complex item: {{{#fold [[Click here]] [[Hidden: {{{#code #lang="rust" fn main() { println!("Deep!"); } }}}]] }}} ]]
[[Final chapter with ^^superscript^^ text]]
}}}