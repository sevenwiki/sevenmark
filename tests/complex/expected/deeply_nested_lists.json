[
  {
    "ListElement": {
      "location": {
        "start": 0,
        "end": 354
      },
      "kind": "1",
      "parameters": {
        "1": {
          "location": {
            "start": 8,
            "end": 12
          },
          "key": "1",
          "value": []
        },
        "style": {
          "location": {
            "start": 12,
            "end": 38
          },
          "key": "style",
          "value": [
            {
              "Text": {
                "location": {
                  "start": 20,
                  "end": 36
                },
                "content": "margin-left:20px"
              }
            }
          ]
        }
      },
      "content": [
        {
          "parameters": {},
          "content": [
            {
              "Bold": {
                "location": {
                  "start": 40,
                  "end": 53
                },
                "content": [
                  {
                    "Text": {
                      "location": {
                        "start": 42,
                        "end": 51
                      },
                      "content": "Chapter 1"
                    }
                  }
                ]
              }
            },
            {
              "Text": {
                "location": {
                  "start": 53,
                  "end": 67
                },
                "content": ": Introduction"
              }
            }
          ]
        },
        {
          "parameters": {},
          "content": [
            {
              "Text": {
                "location": {
                  "start": 72,
                  "end": 84
                },
                "content": "Subsection: "
              }
            },
            {
              "ListElement": {
                "location": {
                  "start": 84,
                  "end": 150
                },
                "kind": "a",
                "parameters": {
                  "a": {
                    "location": {
                      "start": 92,
                      "end": 96
                    },
                    "key": "a",
                    "value": []
                  }
                },
                "content": [
                  {
                    "parameters": {},
                    "content": [
                      {
                        "Text": {
                          "location": {
                            "start": 98,
                            "end": 111
                          },
                          "content": "Point A with "
                        }
                      },
                      {
                        "Italic": {
                          "location": {
                            "start": 111,
                            "end": 121
                          },
                          "content": [
                            {
                              "Text": {
                                "location": {
                                  "start": 112,
                                  "end": 120
                                },
                                "content": "emphasis"
                              }
                            }
                          ]
                        }
                      }
                    ]
                  },
                  {
                    "parameters": {},
                    "content": [
                      {
                        "Text": {
                          "location": {
                            "start": 126,
                            "end": 139
                          },
                          "content": "Point B with "
                        }
                      },
                      "TimeNow"
                    ]
                  }
                ]
              }
            },
            {
              "Text": {
                "location": {
                  "start": 150,
                  "end": 151
                },
                "content": " "
              }
            }
          ]
        },
        {
          "parameters": {},
          "content": [
            {
              "Bold": {
                "location": {
                  "start": 156,
                  "end": 169
                },
                "content": [
                  {
                    "Text": {
                      "location": {
                        "start": 158,
                        "end": 167
                      },
                      "content": "Chapter 2"
                    }
                  }
                ]
              }
            },
            {
              "Text": {
                "location": {
                  "start": 169,
                  "end": 186
                },
                "content": ": Advanced Topics"
              }
            }
          ]
        },
        {
          "parameters": {},
          "content": [
            {
              "Text": {
                "location": {
                  "start": 191,
                  "end": 205
                },
                "content": "Complex item: "
              }
            },
            {
              "FoldElement": {
                "location": {
                  "start": 205,
                  "end": 303
                },
                "parameters": {},
                "content": [
                  {
                    "parameters": {},
                    "content": [
                      {
                        "Text": {
                          "location": {
                            "start": 216,
                            "end": 226
                          },
                          "content": "Click here"
                        }
                      }
                    ]
                  },
                  {
                    "parameters": {},
                    "content": [
                      {
                        "Text": {
                          "location": {
                            "start": 231,
                            "end": 239
                          },
                          "content": "Hidden: "
                        }
                      },
                      {
                        "CodeElement": {
                          "location": {
                            "start": 239,
                            "end": 297
                          },
                          "language": [
                            {
                              "Text": {
                                "location": {
                                  "start": 255,
                                  "end": 259
                                },
                                "content": "rust"
                              }
                            }
                          ],
                          "content": [
                            {
                              "Text": {
                                "location": {
                                  "start": 261,
                                  "end": 271
                                },
                                "content": "fn main() "
                              }
                            },
                            {
                              "Text": {
                                "location": {
                                  "start": 271,
                                  "end": 272
                                },
                                "content": "{"
                              }
                            },
                            {
                              "Text": {
                                "location": {
                                  "start": 272,
                                  "end": 292
                                },
                                "content": " println!(\"Deep!\"); "
                              }
                            },
                            {
                              "Text": {
                                "location": {
                                  "start": 292,
                                  "end": 293
                                },
                                "content": "}"
                              }
                            },
                            {
                              "Text": {
                                "location": {
                                  "start": 293,
                                  "end": 294
                                },
                                "content": " "
                              }
                            }
                          ]
                        }
                      }
                    ]
                  }
                ]
              }
            },
            {
              "Text": {
                "location": {
                  "start": 303,
                  "end": 304
                },
                "content": " "
              }
            }
          ]
        },
        {
          "parameters": {},
          "content": [
            {
              "Text": {
                "location": {
                  "start": 309,
                  "end": 328
                },
                "content": "Final chapter with "
              }
            },
            {
              "Superscript": {
                "location": {
                  "start": 328,
                  "end": 343
                },
                "content": [
                  {
                    "Text": {
                      "location": {
                        "start": 330,
                        "end": 341
                      },
                      "content": "superscript"
                    }
                  }
                ]
              }
            },
            {
              "Text": {
                "location": {
                  "start": 343,
                  "end": 348
                },
                "content": " text"
              }
            }
          ]
        }
      ]
    }
  }
]